imi/src/main/java/org/motechproject/nms/imi/component/CallRescheduler.java

```
@@ -0,0 +1,97 @@
+                        subscription.getSubscriber().getCircle(),
+                        subscription.getOrigin().getCode()
+                ));
+            } else {
+                //we've already rescheduled this call, let's see if it needs to be re-rescheduled
+
+                if (subscription.getSubscriptionPack().retryCount() == 1) {
+                    //This message should only be retried once, so let's delete it from the CallRetry table
+                    LOGGER.info("Not re-rescheduling single-retry msisdn {} subscription {}: max retry exceeded",
+                            cdr.getMsisdn(), requestId.getSubscriptionId());
+                    callRetryDataService.delete(callRetry);
+                    return;
+                }
+
+                if (callRetry.getCallStage() == CallStage.RETRY_LAST) {
+                    //This message has been re-scheduled for the last (3rd) time, let's delete it from the CallRetry
+                    //table
+                    LOGGER.info("Not re-rescheduling multiple-retry msisdn {} subscription {}: max retry exceeded",
+                            cdr.getMsisdn(), requestId.getSubscriptionId());
+                    callRetryDataService.delete(callRetry);
+                    return;
+                }
+
+                //re-reschedule the call
+                LOGGER.info("re-rescheduling msisdn {} subscription {}", cdr.getMsisdn(), requestId.getSubscriptionId());
+
+                //update the callStage
+                callRetry.setCallStage((callRetry.getCallStage() == CallStage.RETRY_1) ? CallStage.RETRY_2 :
+                        CallStage.RETRY_LAST);
+                //increment the day of the week
+                callRetry.setDayOfTheWeek(callRetry.getDayOfTheWeek().nextDay());
+                //update the CallRetry record
+                callRetryDataService.update(callRetry);
+            }
+        } catch (Exception e) {
```

This feels a little risky right? Catching and swallowing a general exception? Ok to swallow known exceptions but we should crash on unknowns